src/addons/presenter.js

```diff
@@ -248,33 +251,44 @@ class PresenterMediator extends React.PureComponent {
     return this.model.value
   }
 
+  // Do not allow transfer across repos. Check to for inheritence by comparing
+  // the common history object shared between repos
   hasParent(): boolean {
-    // Do not allow transfer across repos. Check to for inheritence by comparing
-    // the common history object shared between repos
+    // No need to check if there isn't a parent context setup
+    if (!this.context.parent) {
+      return false
+    }
+
     return get(this.repo, 'history') === get(this.context, ['repo', 'history'])
   }
```

Comment:
so you're using history as a unique identifier of sorts? It's confusing to me to see history used in comparison here, doesn't seem like it has a logical reason to be in this area of code. But my understanding is that the only problem this is solving is the fact that an action should be caught by the closest repo available, and then passed up the parent chain if it doesn't catch it, perhaps i'm missing something about the problem.